projects/plugins/jetpack/class.jetpack-gutenberg.php

```diff
@@ -679,38 +681,44 @@ public static function enqueue_block_editor_assets() {
-				'available_blocks' => self::get_availability(),
-				'jetpack'          => array(
-					'is_active'                 => Jetpack::is_connection_ready(),
-					'is_current_user_connected' => $is_current_user_connected,
-					/** This filter is documented in class.jetpack-gutenberg.php */
-					'enable_upgrade_nudge'      => apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false ),
-					'is_private_site'           => '-1' === get_option( 'blog_public' ),
-					'is_coming_soon'            => ( function_exists( 'site_is_coming_soon' ) && site_is_coming_soon() ) || (bool) get_option( 'wpcom_public_coming_soon' ),
-					'is_offline_mode'           => $status->is_offline_mode(),
-					/**
-					 * Enable the RePublicize UI in the block editor context.
-					 *
-					 * @module publicize
-					 *
-					 * @since 10.3.0
-					 *
-					 * @param bool true Enable the RePublicize UI in the block editor context. Defaults to true.
-					 */
-					'republicize_enabled'       => apply_filters( 'jetpack_block_editor_republicize_feature', true ),
-				),
-				'siteFragment'     => $status->get_site_suffix(),
-				'adminUrl'         => esc_url( admin_url() ),
-				'tracksUserData'   => $user_data,
-				'wpcomBlogId'      => $blog_id,
-				'allowedMimeTypes' => wp_get_mime_types(),
-				'siteLocale'       => str_replace( '_', '-', get_locale() ),
-			)
+		$initial_state = array(
+			'available_blocks' => self::get_availability(),
+			'jetpack'          => array(
+				'is_active'                 => Jetpack::is_connection_ready(),
+				'is_current_user_connected' => $is_current_user_connected,
+				/** This filter is documented in class.jetpack-gutenberg.php */
+				'enable_upgrade_nudge'      => apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false ),
+				'is_private_site'           => '-1' === get_option( 'blog_public' ),
+				'is_coming_soon'            => ( function_exists( 'site_is_coming_soon' ) && site_is_coming_soon() ) || (bool) get_option( 'wpcom_public_coming_soon' ),
+				'is_offline_mode'           => $status->is_offline_mode(),
+				/**
+				 * Enable the RePublicize UI in the block editor context.
+				 *
+				 * @module publicize
+				 *
+				 * @since 10.3.0
+				 * @deprecated $$next_version$$ This is a feature flag that is no longer used.
+				 *
+				 * @param bool true Enable the RePublicize UI in the block editor context. Defaults to true.
+				 */
+				'republicize_enabled'       => apply_filters( 'jetpack_block_editor_republicize_feature', true ),
+			),
+			'siteFragment'     => $status->get_site_suffix(),
+			'adminUrl'         => esc_url( admin_url() ),
+			'tracksUserData'   => $user_data,
+			'wpcomBlogId'      => $blog_id,
+			'allowedMimeTypes' => wp_get_mime_types(),
+			'siteLocale'       => str_replace( '_', '-', get_locale() ),
 		);
+
+		if ( ( new Modules() )->is_active( 'publicize' ) ) {
+			$initial_state['social'] = array(
+				'sharesData'  => publicize_init()->get_publicize_shares_info( $blog_id ),
```

Comment:
If I'm reading this right, any Jetpack site (including simple sites) will call our endpoint every time the editor gets loaded, right? I think we should add a feature flag check on the WPCOM side to avoid causing big spikes.

Reply:
Yes, that's true, although really it's sort of necessary because it's the endpoint which provides the data as to whether the feature flag is enabled.

I have been wondering about the performance of this. It will affect Jetpack sites which have the `publicize` module enabled, so it's not all Jetpack sites, but it is all simple sites.

We could cache the information about whether the share limit is enabled in the site information that is refreshed on the Jetpack Dashboard, but eventually everyone will have it enabled and will be calling the endpoint to get the current status.

> I think we should add a feature flag check on the WPCOM side to avoid causing big spikes.

Do you mean to prevent spikes in queries for the number of shares?

Reply:
> really it's sort of necessary because it's the endpoint which provides the data as to whether the feature flag is enabled.

Can we give back dummy data if the feature flag is disabled (since we're not using it anyway)?

> Do you mean to prevent spikes in queries for the number of shares?

Yes, basically my question right above. If we're not going to use that data anyway, maybe we can return a bunch of zeroes instead and slowly ramp it up when we start enabling it for more sites.

Reply:
Yes, we could do that. It solves the issue for now, but I think we would need to work out the caching we've been discussing previously. I'm not sure everyone is aware of the amount of queries this could generate. 

Am I right in thinking that we'll have the same problem with the classic editor?

Reply:
Not in the current state (my PR adds it to the Jetpack Social plugin only), but yes once we move it to the package we will.